packages/eslint-plugin/src/rules/no-new-in-loop.ts

```diff
@@ -0,0 +1,28 @@
+import type { Rule } from 'eslint'
+import * as ESTree from 'estree'
+
+const rule: Rule.RuleModule = {
+  meta: {
+    messages: {
+      noNew:
+        'Instantiating new objects in the frame loop can cause performance problems. Instead, create in the parent scope.',
```

Comment:
Check the wording please! + description

Reply:
Maybe "create once in a useMemo (ignoring the fact that useState has lazy init) or create a single, shared reference outside of the component".

Reply:
Cheers will wordsmith in!